### ISSUE TITLE
Rule Request: prefer 'joined()' over 'joined(separator: "")'

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#1093]
since that's the default separator.

**Comment by aamctustwo** [comment]
Does SourceKitten provide sufficient detail to make this rule more general? In other words, detect any instance where a function is being used with its default arguments.

**Comment by marcelofabri** [comment]
I don't think so, we'd have to keep a list with known functions and default parameters.

**Comment by jpsim** [comment]
> Does SourceKitten provide sufficient detail to make this rule more general?

No, it doesn't, because binary Swift modules don't actually contain that information 😱 

For example, looking at the generated interface for the Swift standard library in Xcode itself, you get this:

```swift
extension Array where Element == String {
    public func joined(separator: String = default) -> String
}
```

Notice the `default` in the place where you'd expect to see `""`.

I actually _think_ (I could be wrong) that this has something to do with Swift's [planned ABI stability](https://github.com/apple/swift/blob/master/docs/ABI.rst). But I'm not sure.

**Comment by marcelofabri** [comment]
Closed by @ornithocoder in #1746 🎉
